src/rgw/rgw_op.cc

```diff
@@ -5276,6 +5276,30 @@ void RGWCompleteMultipart::execute()
   meta_obj.set_in_extra_data(true);
   meta_obj.index_hash_source = s->object.name;
 
+  rgw_pool meta_pool;
```

Comment:
could you add a comment above this block? something simple like
> take a cls lock on meta_obj to prevent racing completions (or retries) from deleting the parts